cmd/compare.go

```diff
@@ -254,3 +322,22 @@ func validateTypes(old *schema.TypeSpec, new *schema.TypeSpec, prefix string) (v
 func formatName(provider, s string) string {
 	return strings.ReplaceAll(strings.TrimPrefix(s, fmt.Sprintf("%s:", provider)), ":", ".")
 }
+
+type set[T comparable] struct{ m map[T]struct{} }
```

Comment:
Curious, why do we need the outer `struct` here?

Reply:
Two reasons:
1. Because it makes using the `Has` method more intuitive than a `_, has := s["value"]` check. This provides better encapsulation on the underlying representation.
2. How methods work on `nil`  values of typed nillable types (like `map[K]V`) is super weird. You cannot have a method declared as `func (m *MapAlias) Name()`. Adding a `struct` wrapper allows me to work around this. Should I need to, I could add an `Insert` method that would work on a zero-valued `set[T]`.